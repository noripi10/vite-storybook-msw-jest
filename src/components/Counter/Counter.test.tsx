import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Counter } from './';

describe('Counter', () => {
  let CounterElement: HTMLElement;
  beforeEach(() => {
    const { asFragment, container } = render(<Counter />);
    CounterElement = container;
    expect(asFragment()).toMatchSnapshot();
  });

  test('初期化', () => {
    const count = CounterElement.querySelector('strong');
    expect(count?.innerHTML).toBe('0');
  });

  test('インクリメント', async () => {
    const count = screen.getByTestId('count');
    const button = screen.getByRole('button', { name: 'count up' });
    const countOver2Init = screen.queryByText(/count is now:/);

    expect(countOver2Init).not.toBeInTheDocument();

    userEvent.click(button);
    userEvent.click(button);
    userEvent.click(button);

    await waitFor(() => expect(count.innerHTML).toBe('3'));

    const countOver2 = screen.queryByText(/count is now:/);
    expect(countOver2).toBeInTheDocument();
  });

  test('デクリメント', async () => {
    const count = screen.getByTestId('count');
    const button = screen.getByRole('button', { name: 'count down' });
    const countOver2Init = screen.queryByText(/count is now:/);

    expect(countOver2Init).not.toBeInTheDocument();

    userEvent.click(button);
    userEvent.click(button);
    userEvent.click(button);

    await waitFor(() => expect(count.innerHTML).toBe('-3'));

    const countOver2 = screen.queryByText(/count is now:/);
    expect(countOver2).not.toBeInTheDocument();
  });
});
