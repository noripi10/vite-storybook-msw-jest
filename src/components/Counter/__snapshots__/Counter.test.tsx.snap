// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter インクリメント 1`] = `
<DocumentFragment>
  <div>
    <div>
      <strong
        data-testid="count"
      >
        0
      </strong>
    </div>
    <button>
      count down
    </button>
    <button>
      count up
    </button>
  </div>
</DocumentFragment>
`;

exports[`Counter デクリメント 1`] = `
<DocumentFragment>
  <div>
    <div>
      <strong
        data-testid="count"
      >
        0
      </strong>
    </div>
    <button>
      count down
    </button>
    <button>
      count up
    </button>
  </div>
</DocumentFragment>
`;

exports[`Counter 初期化 1`] = `
<DocumentFragment>
  <div>
    <div>
      <strong
        data-testid="count"
      >
        0
      </strong>
    </div>
    <button>
      count down
    </button>
    <button>
      count up
    </button>
  </div>
</DocumentFragment>
`;
